packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -168,7 +168,7 @@ export class Bundling {
       // Always rename dist file to index.js because Lambda doesn't support filenames
       // with multiple dots and we can end up with multiple dots when using automatic
       // entry lookup
-      `mv ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${distFile} ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/index.js`,
+      distFile !== 'index.js' ? `mv ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${distFile} ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/index.js` : '',
```

Comment:
Would it be less error prone to just have parcel output as index.js? https://github.com/parcel-bundler/parcel/issues/884

Doesn't look like using the entry name has any value and we could remove the parsing on line 155.

```suggestion
      '--out-file index.js'
```

Reply:
Unfortunately `--out-file` is a Parcel v1 option, not v2. In v2 you need to configure it in the `package.json` but I want to avoid this to allow #9632

Reply:
Ah, I see. Parcel seems like a pile.